bindings/rascal/neighbourlist/structure_manager.py

```diff
@@ -235,4 +255,13 @@ def add_center_atoms_mask_species(frame, species_select=[],
     else:
         raise ValueError(
                 "Species blacklist must be either all string or all int")
-    add_center_atoms_mask_by_id(frame, id_select, id_blacklist)
+    if 'center_atoms_mask' not in frame.arrays:
+        # add a default mask
+        if species_select:
+            old_mask = np.zeros((frame.get_number_of_atoms(),), dtype='bool')
+        else:
+            old_mask = np.ones((frame.get_number_of_atoms(),), dtype='bool')
+    else:
+        old_mask = frame.arrays['center_atoms_mask']
+    mask = (old_mask | id_select) & ~id_blacklist
```

Comment:
Could you add a comment here, I guess not everyone is familiar with boolean binary operations

Reply:
These are standard Python operators and the result (and logic) is clearly described in the docstring. I don't believe in comments that just repeat the code, which is what this would have to be.

Reply:
If you insist, I can add an RTFM to here: https://docs.scipy.org/doc/numpy/reference/ufuncs.html#comparison-functions

But I really don't think it's necessary.

Reply:
These operators are not often used in Python, the preferred style to use `and`, `or`, `not`. This is not possible here for the reason showed in your link, but might still confuse newcomers coming to this code.

I would still at least add something like 'combine all masks using bitwise operators, since standard Python operators should not be used with numpy array (<link>)'.

Reply:
If a newcomer is digging into the code and sees this line, I'd better hope they already know the difference between `and` and `&` when it comes to NumPy arrays (practically anyone who's worked with numpy before has seen that iconic "The truth value of an array is ambiguous..." error).

Reply:
Seriously, I'm not doing the job of standard documentation. The comment you suggested belongs in the Numpy docs, not in ours.

Reply:
I was slightly confused by reading this, and it is not much work for something that can help newcomers to the code.

But if you really don't want to, I'll be fine 😃